python/cudf_polars/cudf_polars/dsl/translate.py

```diff
@@ -114,51 +211,51 @@ def _(
         skip_rows,
         n_rows,
         row_index,
-        translate_named_expr(visitor, n=node.predicate)
+        translate_named_expr(translator, n=node.predicate)
         if node.predicate is not None
         else None,
     )
 
 
 @_translate_ir.register
 def _(
-    node: pl_ir.Cache, visitor: NodeTraverser, schema: dict[str, plc.DataType]
+    node: pl_ir.Cache, translator: Translator, schema: dict[str, plc.DataType]
 ) -> ir.IR:
-    return ir.Cache(schema, node.id_, translate_ir(visitor, n=node.input))
+    return ir.Cache(schema, node.id_, translator.translate_ir(n=node.input))
 
 
 @_translate_ir.register
 def _(
-    node: pl_ir.DataFrameScan, visitor: NodeTraverser, schema: dict[str, plc.DataType]
+    node: pl_ir.DataFrameScan, translator: Translator, schema: dict[str, plc.DataType]
 ) -> ir.IR:
     return ir.DataFrameScan(
         schema,
         node.df,
         node.projection,
-        translate_named_expr(visitor, n=node.selection)
+        translate_named_expr(translator, n=node.selection)
         if node.selection is not None
         else None,
     )
 
 
 @_translate_ir.register
 def _(
-    node: pl_ir.Select, visitor: NodeTraverser, schema: dict[str, plc.DataType]
+    node: pl_ir.Select, translator: Translator, schema: dict[str, plc.DataType]
 ) -> ir.IR:
-    with set_node(visitor, node.input):
-        inp = translate_ir(visitor, n=None)
-        exprs = [translate_named_expr(visitor, n=e) for e in node.expr]
+    with set_node(translator.visitor, node.input):
+        inp = translator.translate_ir(n=None)
+        exprs = [translate_named_expr(translator, n=e) for e in node.expr]
     return ir.Select(schema, inp, exprs, node.should_broadcast)
 
 
 @_translate_ir.register
 def _(
-    node: pl_ir.GroupBy, visitor: NodeTraverser, schema: dict[str, plc.DataType]
+    node: pl_ir.GroupBy, translator: Translator, schema: dict[str, plc.DataType]
 ) -> ir.IR:
-    with set_node(visitor, node.input):
-        inp = translate_ir(visitor, n=None)
-        aggs = [translate_named_expr(visitor, n=e) for e in node.aggs]
-        keys = [translate_named_expr(visitor, n=e) for e in node.keys]
+    with set_node(translator.visitor, node.input):
+        inp = translator.translate_ir(n=None)
+        aggs = [translate_named_expr(translator, n=e) for e in node.aggs]
+        keys = [translate_named_expr(translator, n=e) for e in node.keys]
     return ir.GroupBy(
         schema,
         inp,
```

Comment:
From a developers perspective, I'd like to see all of them so I can know which arg types to implement next. But I can see how this could get pretty verbose from a users perspective. I think we can address both in a follow-up where we organize there errors in user-friendly way (eg. a table). I can open an issue to discuss how we'd want this to look.

Reply:
Thanks!